compiler/src/iree/compiler/Codegen/LLVMGPU/test/tensor_pad.mlir

```diff
@@ -0,0 +1,66 @@
+      ^bb0(%arg2: f32, %arg3: f32):
+        linalg.yield %arg2 : f32
+      } -> tensor<?x32xf32>
+      flow.dispatch.tensor.store %9, %1, offsets = [%arg0, %arg1], sizes = [%4, 32], strides = [1, 1] : tensor<?x32xf32> -> !flow.dispatch.tensor<writeonly:48x32xf32>
+    }
+  }
+  return
+}
+
+// CHECK-LABEL:  func.func @transpose_no_align_dispatch_0_generic_48x32
+//       CHECK:  %[[CST:.*]] = arith.constant 0.000000e+00 : f32
+//       CHECK:  %[[C48:.*]] = arith.constant 48 : index
+//       CHECK:  %[[C32:.*]] = arith.constant 32 : index
+//       CHECK:  %[[C0:.*]] = arith.constant 0 : index
+//       CHECK:  %[[D0:.*]] = hal.interface.binding.subspan set(0) binding(0) type(storage_buffer) offset(%[[C0]]) alignment(64) : !flow.dispatch.tensor<readonly:32x48xf32>
+//       CHECK:  %[[D1:.*]] = hal.interface.binding.subspan set(0) binding(1) type(storage_buffer) offset(%[[C0]]) alignment(64) : !flow.dispatch.tensor<writeonly:48x32xf32>
+//       CHECK:  %[[WORKGROUP_ID_X:.*]] = hal.interface.workgroup.id[0] : index
+//       CHECK:  %[[WORKGROUP_COUNT_X:.*]] = hal.interface.workgroup.count[0] : index
+//       CHECK:  %[[WORKGROUP_ID_Y:.*]] = hal.interface.workgroup.id[1] : index
+//       CHECK:  %[[WORKGROUP_COUNT_Y:.*]] = hal.interface.workgroup.count[1] : index
+//       CHECK:  %[[D2:.*]] = affine.apply #{{.*}}(){{\[}}%[[WORKGROUP_ID_Y]]]
+//       CHECK:  %[[D3:.*]] = affine.apply #{{.*}}(){{\[}}%[[WORKGROUP_COUNT_Y]]]
+//       CHECK:  scf.for %[[ARG0:.*]] = %[[D2]] to %[[C48]] step %[[D3]] {
+//       CHECK:    %[[D4:.*]] = affine.min #{{.*}}(%[[ARG0]])
+//       CHECK:    %[[D5:.*]] = affine.apply #{{.*}}(){{\[}}%[[WORKGROUP_ID_X]]]
+//       CHECK:    %[[D6:.*]] = affine.apply #{{.*}}(){{\[}}%[[WORKGROUP_COUNT_X]]]
+//       CHECK:    scf.for %[[ARG1:.*]] = %[[D5]] to %[[C32]] step %[[D6]] {
+//       CHECK:      %[[D7:.*]] = flow.dispatch.tensor.load %[[D1]], offsets = {{\[}}%[[ARG0]], %[[ARG1]]], sizes = {{\[}}%[[D4]], 32], strides = [1, 1] : !flow.dispatch.tensor<writeonly:48x32xf32> -> tensor<?x32xf32>
+//       CHECK:      %[[D8:.*]] = flow.dispatch.tensor.load %[[D0]], offsets = {{\[}}%[[ARG1]], %[[ARG0]]], sizes = [32, %[[D4]]], strides = [1, 1] : !flow.dispatch.tensor<readonly:32x48xf32> -> tensor<32x?xf32>
+//       CHECK:      %[[D9:.*]] = affine.apply #map2(%[[D4]])
+//       CHECK:      %[[D10:.*]] = tensor.pad %[[D8]] low{{\[}}%[[C0]], %[[C0]]] high{{\[}}%[[C0]], %[[D9]]] {
+//       CHECK:      ^bb0(%[[ARG2:.*]]: index, %[[ARG3:.*]]: index):
+//       CHECK:        tensor.yield %[[CST]] : f32
+//       CHECK:      } : tensor<32x?xf32> to tensor<32x32xf32>
+//       CHECK:      %[[D11:.*]] = tensor.pad %[[D7]] low{{\[}}%[[C0]], %[[C0]]] high{{\[}}%[[D9]], %[[C0]]] {
```

Comment:
I thought we could avoid padding the output?

Reply:
I'm saving that for a followup. 

I'm also worried that always defaulting the output to an init tensor may be incorrect in certain situations, for instance if this linalg is fused with a producer/consumer that actually makes use of the accumulator?

A better solution might be to extend the pattern such that it looks up the source op to ensure it was created by an `init_tensor` before not using pad?